add survey.js@0.12.1 w/ npm auto-update and manually add old versions

sufuf3 opened the pull request:
Pull request for issue: #9771 
Related issue(s): # #
@kennynaoh please help me review this PR, thank you.
Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.

# Profile of the lib
 * Git repository (required): http://github.com/andrewtelnov/surveyjs
 * Official website (optional, not the repository): http://surveyjs.org/
 * NPM package url (optional): https://www.npmjs.com/package/survey-knockout
 * License and its reference: MIT
 * GitHub / Bitbucket popularity (required):
   - Count of stars: 101
   - Count of watchers: 18
   - Count of forks: 48
 * NPM download stats (optional):
   - Downloads in the last day: 4
   - Downloads in the last week: 172
   - Downloads in the last month: 700

# Essential checklist
 * [ ] I'm the author of this library
   * [ ] I would like to add link to the page of this library on CDNJS on website / readme
 * [x] This lib was not found on cdnjs repo
 * [x] No already exist / duplicated issue and PR
 * [x] The lib has notable popularity
   * [x] More than 100 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
   * [x] More than 500 downloads stats per month on npm registry
 * [ ] Project has public repository on famous online hosting platform (or been hosted on npm)

# Auto-update checklist
 * [x] Has valid tags for each versions (for git auto-update)
 * [x] Auto-update setup
 * [x] Auto-update target/source is valid.
 * [x] Auto-update filemap is correct.

# Git commit checklist
 * [x] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
 * [x] The parent of the commit(s) in the PR is not old than 3 days.
 * [x] Pull request is sending from a non-master branch with meaningful name.
 * [x] Separate unrelated changes into different commits.
 * [x] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
 * [x] Close corresponding issue in commit message
 * [x] Mention related issue(s), people in commit message, comment.

PeterDaveHello commented on ajax/libs/survey.js/package.json:
```diff
@@ -0,0 +1,32 @@
+    {
+      "basePath": "dist",
+      "files": [
+        "survey.ko*"
```
Let's use `"**/survey.ko*"`

sufuf3 replied:
I modified it. @kennynaoh please help me review this PR, thank you.

kennynaoh commented on ajax/libs/survey.js/package.json:
```diff
@@ -0,0 +1,32 @@
+    "Knockout"
+  ],
+  "filename": "survey.ko.min.js",
+  "homepage": "https://surveyjs.org/",
```
Let's use `http` because the website doesn't support `https`

sufuf3 replied:
I modified it. @kennynaoh please help me review this PR again, thank you.

PeterDaveHello commented:
#9771 also mentions survey-react, but not only survey-knockout

sufuf3 commented:
@kennynaoh please help me review this PR, thank you.

sufuf3 commented:
@kennynaoh please help me review this PR again, thank you.

sufuf3 commented:
ping @kennynaoh

kennynaoh commented:
I think we should send four PR in this case instead of four commits, should we? @PeterDaveHello

PeterDaveHello commented:
It's fine.

sufuf3 commented:
@kennynaoh should I just rebase master, then it well be OK?

kennynaoh commented:
@sufuf3 Please minify `survey.css`.

sufuf3 commented:
@kennynaoh please help me review this PR again, thank you.

sufuf3 commented:
ping @kennynaoh

sufuf3 commented:
@kennynaoh I fixed the structure of these old versions, added the new lib survey-vue, and updated the new version.
Please help me review this PR again, thank you.

@cdnjs/intern3 please help me review this PR, thank you.

kennynaoh commented:
There is only css file in the v0.11.0 of survey-knockout, could you confirm with author? Thanks!

sufuf3 commented:
@kennynaoh I added the v0.11.0 of survey-knockout files from https://github.com/surveyjs/surveyjs/releases/tag/0.11.
Please help me review this PR again, thank you.
@cdnjs/intern3 please help me review this PR, thank you.

kennynaoh commented:
LGTM

PeterDaveHello commented:
@extend1994, @hare1039, and @clairetsai818 ?

pvnr0082t commented on ajax/libs/survey.js/package.json:
```diff
@@ -0,0 +1,32 @@
+    "name": "andrewtelnov",
+    "email": "andrew.telnov@gmail.com"
+  },
+  "npmName": "survey-knockout",
```
`survey.js` also fetch the files from `survey-knockout`?

sufuf3 commented:
@pvnr0082t @extend1994 sorry about that I shouldn't host the `survey.js`.
It's not a lib form git repo or npm, so I remove it.
Please help me review this PR again, thank you.

maruilian11 commented:
I think we had better not close the issue by one of this commit, you can close it manually  after this PR being merged.

sufuf3 commented:
@maruilian11 you meaning that I should change the commit message?
from 
```
add survey-vue@0.12.1 w/ npm auto-update  
close #9771, cc @andrewtelnov
```
to
```
Add survey-vue@0.12.1 w/ npm auto-update  
cc #9771, cc @andrewtelnov
```

sufuf3 commented:
ping @maruilian11 @pvnr0082t

PeterDaveHello commented:
I got strange situation here, the branch I fetched locally has different orders, not like what do I see on GitHub.

@sufuf3 which one is correct? I mean the same as your local.

What I see on GitHub:
![image](https://cloud.githubusercontent.com/assets/3691490/23551330/f1845b9a-0050-11e7-919f-c3273cce9ef0.png)

On my local:
![2017-03-03 8 34 25](https://cloud.githubusercontent.com/assets/3691490/23551339/f8269eb8-0050-11e7-98e6-a6d45b6926e3.png)

(PS: the order should be reversed.)

maruilian11 commented:
@sufuf3 yes, please.

sufuf3 commented:
@PeterDaveHello on my local is same as yours.
I don't know how to change the commit orders on GitHub.

PeterDaveHello commented:
@sufuf3 Same order and same hashes?

sufuf3 commented:
Yes
![image](https://cloud.githubusercontent.com/assets/8349587/23580298/81633c2a-0139-11e7-84a5-496b7b794ef3.png)

sufuf3 commented:
@PeterDaveHello  Yes
![image](https://cloud.githubusercontent.com/assets/8349587/23580298/81633c2a-0139-11e7-84a5-496b7b794ef3.png)

PeterDaveHello commented:
@sufuf3 Then we should report a bug to GitHub.

sufuf3 commented:
@maruilian11 I changed the commit message.
Please help me review this PR again, thank you.

maruilian11 commented:
please update the title

sufuf3 commented:
@maruilian11 Is the new title OK?
Because this PR include many libraries from npm, and not all the libraries manually add the same old versions.
So I use `add survey.js@0.12.1 w/ npm auto-update and manually add old versions` as this title.

If there is any suggest, please let me know.
Thank you.

maruilian11 commented:
@sufuf3 I think this PR is good now, please rebase the branch with our latest master.

sufuf3 commented:
@maruilian11 I rebased the branch with the latest master.
Please help me review this PR again, thank you.
